tests/run.sh

```
@@ -299,5 +299,90 @@ else
+# Primitives use hue-based names (--color-neutral-*, --color-blue-*, etc.)
+# Semantic roles do not — catch any --color-* that isn't a primitive scale
+semantic_re = re.compile(
+    r'^\s*(--color-(?!neutral|blue|green|red|amber|cyan)[a-z][\w-]*):', re.MULTILINE
+)
+hits = semantic_re.findall(layer_body)
+if hits:
+    print(f"ERROR: semantic --color-* roles found inside @layer foundation: {hits[:5]}")
+    sys.exit(1)
+print("OK")
+PYEOF
+if [ $? -ne 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)